### PR TITLE
Replace validate_apache_loglevel() with data type

### PR DESCRIPTION
**Pull request by alexjfisher**
The function was introduced before Puppet 4 was released.  It was
originally created to reduce duplicated use of stdlib's `validate_re`.

See https://github.com/puppetlabs/puppetlabs-apache/pull/1097

### CODE REVIEW

File: spec/type_aliases/loglevel_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    'warn ssl:info',
+    'warn mod_ssl.c:info',
+    'warn mod_ssl.c:info',
+    'warn ssl_module:info',
+    'trace4',
+  ].each do |allowed_value|
+    it { is_expected.to allow_value(allowed_value) }
+  end
+
+  [
```
Comment (ekohl) [inline spec/type_aliases/loglevel_spec.rb:15]: How about `xemerg` to test if `^` at the start is needed?

File: types/loglevel.pp [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# @summary As per http://httpd.apache.org/docs/current/mod/core.html#loglevel
```
Comment (ekohl) [inline types/loglevel.pp:1]: Would `@see` be useful for the URL?

File: spec/type_aliases/loglevel_spec.rb (hunk 1 above)
Comment (alexjfisher) [reply spec/type_aliases/loglevel_spec.rb:15]: The regex is copied from the the function and is very lax and could be better.  I thought about trying to make it stricter, but decided that a change like that might be suited to a separate PR.

As things stand a `^` at the beginning would be bad. The apache docs describe the allowed syntax as...

> Syntax
> LogLevel [module:]level [module:level] ...

I think in a perfect world, $log_level would have actually accepted an `Array`. I doubt it's worth introducing that sort of change now though.

File: types/loglevel.pp (hunk 2 above)
Comment (alexjfisher) [reply types/loglevel.pp]: I've had a go at making the puppet-strings based docs better. Changes to the REFERENCE.md aren't in this commit as it's very difficult to exclude unrelated changes.  It looks like the REFERENCE.md will be updated as part of the release process.

File: manifests/init.pp [hunk 3]
```diff
@@ -195,8 +195,8 @@
 #   > **Note**: Do not configure this parameter manually without special reason.
 #
 # @param log_level
-#   Changes the error log's verbosity. Valid options are: `alert`, `crit`, `debug`, `emerg`, `error`, 
-#   `info`, `notice` and `warn`.
+#   Configures the apache [LogLevel](https://httpd.apache.org/docs/current/mod/core.html#loglevel) directive
```
Comment (sanfrancrisko) [inline manifests/init.pp:198]:
```suggestion
#   Configures the Apache [LogLevel](https://httpd.apache.org/docs/current/mod/core.html#loglevel) directive
```

**Pull request merged**